janusgraph-core/src/main/java/org/janusgraph/graphdb/query/index/IndexCandidate.java

```diff
@@ -0,0 +1,55 @@
+// Copyright 2020 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.graphdb.query.index;
+
+import org.janusgraph.graphdb.query.condition.Condition;
+import org.janusgraph.graphdb.types.IndexType;
+
+import java.util.Set;
+
+/**
+ * @author Boxuan Li (liboxuan@connect.hku.hk)
+ */
+public class IndexCandidate {
+    private final IndexType index;
+    private final Set<Condition> subCover;
+    private final Object subCondition;
+
+    private double score;
+
+    public IndexCandidate(final IndexType index,
+                          final Set<Condition> subCover,
+                          final Object subCondition) {
+        this.index = index;
+        this.subCover = subCover;
+        this.subCondition = subCondition;
+        // initialize with the worst possible score
+        this.score = Double.NEGATIVE_INFINITY;
```

Comment:
I would put it back to inline initialization because:
1) The field isn't final, it means that we may add a bug if we add another constructor later (i.e. forget to initialize it)
2) I'm not sure about non final fields in sense of initialization. If final field is initialized only once, what about non final field? Will it be initialized with `0.0` at class fields initialization phase and them `Double.NEGATIVE_INFINITY` at constructor initialization phase? Again, not sure about it but if we use inline initialization we guarantee that it will be `Double.NEGATIVE_INFINITY` in both class fields initialization phase and constructor initialization phase.

Reply:
Yes, you're right. Good point.